src/journal/journald.c

```diff
@@ -52,7 +52,7 @@ int main(int argc, char *argv[]) {
                 goto finish;
 
         server_vacuum(&server, false);
-        server_flush_to_var(&server);
+        server_flush_to_var(&server, true);
```

Comment:
well, the PR doesn't fix stuff that is already on disk, but it removes the problem for new installs, as during boot we can reliably schedule tmpfiles before the journal writes to /var/log/journal now. This doesn't mean we magically fix everything up on upgrades, though. And I think that's probably something to do from the postinst update scriptlets of RPMs...

`/run/systemd/journal/flushed` is an internal file managed by journald, if you create it on your own, then it's your own fault really. Sending SIGUSR1 to journald is really the only supported way to create it. And if you do it locally, then that pretty much voids your warranty...

There's currently no easy and safe way to enable persistency, as a simply "mkdir" indeed won't suffice and is racy as the ACLs would only be applied on the next reboot. To provide a nice way for this I figure we could "journalctl --make-persistent" or so which creates the directory under a temporary name first, then applies the right ACLs and then moves it the final name and sends SIGUSR1 to journald to make it flush its stuff there. I think that would work fine and solve the points you raise, wouldn't it?